scarb/src/core/registry/client/mod.rs

```diff
@@ -33,6 +33,23 @@ pub enum RegistryDownload<T> {
     Download(T),
 }
 
+/// Result from uploading files to a registry.
+#[derive(Debug)]
+pub enum RegistryUpload {
+    /// Missing or invalid authentication token.
+    Unauthorized,
+    /// Missing upload permissions or not the package owner.
+    CannotPublish,
+    /// Package version already exists.
+    VersionExists,
+    /// File corrupted during upload.
+    Corrupted,
+    /// Upload failed for other reasons.
+    Failed,
+    /// Upload successful.
+    Success,
+}
```

Comment:
shouldn't all these error cases be kept as `anyhow::Error` objects? you seem to never match on the variants of this struct

Reply:
Why? What would be the advantage of doing this?
I went with this because find it more specific and easy to propagate. It also holds all possible results, not only Errors. But I am willing to change my mind on this, given strong arguments :+1:

Reply:
see @DelevoXDG argument, it'd be easier to just pass error messages from the registry etc.